### PR TITLE
mmcsd: add multi partitions support

### PR DESCRIPTION
**Pull request by gneworld**
## Summary
  include boot0,boot1,gp1,gp2,gp3,gp4,rpmb
## Impact
 mmcsd
## Testing
 cortexM55

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, this pull request does not meet the NuttX requirements.**

**Missing Information:**

* **Summary:**  
    *  Lacks a clear explanation of the change's purpose (fix, update, new feature). 
    *  Doesn't describe the functional area of the code being modified.
    *  Omits details about how the change works.
    *  No issue references provided.
* **Impact:**
    * All impact sections are vague and lack specific details. For instance, "mmcsd" doesn't explain what changed within the mmcsd functionality.
* **Testing:**
    *  Insufficient build host and target details (OS, compiler versions, specific board configurations are needed).
    *  No testing logs provided to demonstrate functionality before and after the change.

**To meet the NuttX requirements, you need to provide:**

* **A comprehensive summary** explaining the "why," "what," and "how" of your changes.
* **Detailed impact descriptions**, specifying the effects on users, builds, hardware, documentation, security, compatibility, etc.
* **Thorough testing information**, including build host and target specifics, as well as testing logs that validate your changes.

File: drivers/mmcsd/mmcsd_sdio.h [hunk 1]
```diff
@@ -82,6 +83,17 @@
                                      MMC_CMD6_INDEX(EXT_CSD_HS_TIMING) | \
                                      MMC_CMD6_MODE(MMC_CMD6_MODE_WRITE_BYTE))
 
+/* Partition type */
+
+#  define MMCSD_PART_UDATA            0
```
Comment (xiaoxiang781216) [inline drivers/mmcsd/mmcsd_sdio.h:88]:
```suggestion
#  define MMCSD_PART_UDATA           0
```

Comment (gneworld) [reply drivers/mmcsd/mmcsd_sdio.h]: done

**Comment by gneworld** [comment]
image
can not reproduce ci error about risc-v-04 locally

**Comment by gneworld** [comment]
image
can not reproduce ci error about risc-v-04 locally

**Pull request closed without merging**